libherokubuildpack/src/buildpack_output/util.rs

```diff
@@ -0,0 +1,223 @@
+use std::fmt::Debug;
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+
+/// Iterator yielding every line in a string. Every line includes existing newline character(s).
+pub(crate) struct LineIterator<'a> {
+    input: &'a str,
+}
+
+impl<'a> LineIterator<'a> {
+    pub(crate) fn from(input: &'a str) -> LineIterator<'a> {
+        LineIterator { input }
+    }
+}
+
+impl<'a> Iterator for LineIterator<'a> {
+    type Item = &'a str;
+
+    #[inline]
+    fn next(&mut self) -> Option<&'a str> {
+        if self.input.is_empty() {
+            return None;
+        }
+
+        let newline_index = self.input.find('\n').map_or(self.input.len(), |i| i + 1);
+
+        let (line, rest) = self.input.split_at(newline_index);
+        self.input = rest;
+        Some(line)
+    }
+}
+
+pub(crate) fn prefix_lines<F: Fn(usize, &str) -> String>(contents: &str, f: F) -> String {
```

Comment:
I found this function confusing, since at first I thought it was the function that would be used by other parts of the `BuildOutput` module, rather than an private helper function only. 

A few things led to this misunderstanding:
- the fact this is `pub(crate) fn` rather than `fn` (it's not used outside of this file)
- the fact that it's ordered first in this file, above `prefix_first_rest_lines` (ie: I thought `prefix_lines` would be calling `prefix_first_rest_lines` not vice versa)

Could you remove the `pub(crate)` and move this function below `prefix_first_rest_lines`?